### PR TITLE
Run schemalint as GitHub action

### PR DESCRIPTION
**Pull request by mogottsch**
This PR adds a GitHub action, which will run `schemalint verify` on the `values.schema.json` file.

Schemalint will verify that the JSON schema
- is valid
- is normalized (consistent indentation and sorting)
- fulfills the [requirements we have on cluster-app schemas](https://github.com/giantswarm/rfc/pull/55)

This is the first time that the validation of the cluster-app schemas is used in production. If you have any problems reach out to Team Rainbow.
Also if you want to discuss the rules implemented, feel free to contribute to the RFC mentioned above.

### CODE REVIEW

**Comment by mogottsch** [comment]
As you can see the schemalint check is currently failing with four errors. I didn't fix them so you can see how it looks when schemalint fails. 
The pipeline will only fail on schemalint errors, which correspond to MUSTs in the [RFC](https://github.com/giantswarm/rfc/pull/55). There are also recommendations that correspond to SHOULDs. 
I am not sure how to include those into a developer's workflow.

**Comment by marians** [comment]
The output appears like this for me:

image

In line 62 it seems as if something output got truncated. Is that something we can influence?

**Comment by mogottsch** [comment]
@marians 
It seems like GitHub actions changes the appearance of the output if it detects `[ERROR]` and `[WARNING]` (I've made a few tests [here](https://github.com/giantswarm/schemalint/tree/test-gh-actions)). I've found nothing about this and therefore I don't know how to deactivate it.
I suggest we remove the `-` in front of `[ERROR]`. That would fix it.

**Comment by marians** [comment]
Let's do that!

**Review by marians — APPROVED** [review]
The changes to come should not block this PR

File: .github/workflows/schemalint.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Schemalint
```
Comment (marians) [inline .github/workflows/schemalint.yaml:1]: As the tool's name is `schemalint`, I would prefer the action name to use the same spelling.

```suggestion
name: schemalint
```

**Comment by primeroz** [comment]
should we wait for https://github.com/giantswarm/cluster-azure/pull/38 before this one ? 

thanks for the work !

**Comment by mogottsch** [comment]
Yes, if we merge #38 then the pipeline should pass.

Have you looked at the [failing job](https://github.com/giantswarm/cluster-azure/actions/runs/4083313198/jobs/7038688524)? Are the error messages understandable and actionable to you?

**Comment by primeroz** [comment]
> Have you looked at the [failing job](https://github.com/giantswarm/cluster-azure/actions/runs/4083313198/jobs/7038688524)? Are the error messages understandable and actionable to you?

yes , it was highlighting the `wrong schema url` so i think :+1: 

was that the only error ? or would we see only the first error with the rest hidden ?

**Comment by mogottsch** [comment]
There were actually 4 errors listed:

```
Errors (4)

- Property '/defaults' must have a title.
- Property '/includeClusterResourceSet' must have a title.
- Property '/machinePools' must have a title.
- Schema must use draft/dialect 'https://json-schema.org/draft/2019-09/schema', but uses 'http://json-schema.org/schema#'.
```

**Comment by marians** [comment]
#38 is merged

**Comment by marians** [comment]
Oh yes, damn. I still left out that one:

```
Array '/machinePools' must specify the schema of its items.
```

**Comment by mogottsch** [comment]
@marians The pipeline is still failing:
```
- Array '/machinePools' must specify the schema of its items.
```
`machinePools` is deprecated. Maybe we should talk about not applying rules when the given property is deprecated.

**Comment by marians** [comment]
> `machinePools` is deprecated. Maybe we should talk about not applying rules when the given property is deprecated.

I think that could have bad consequences, since even a deprecated key can be used, and then there wouldn't be any validation. I'm afraid we have to add the items schema here.

**Comment by mogottsch** [comment]
I'm afraid I'm lacking knowledge about what is a valid item in `machinePools`. 
@primeroz could you take a look at this?

**Comment by mogottsch** [comment]
Nevermind, I've found what I need in `values.yaml` :)

**Comment by primeroz** [comment]
> I'm afraid I'm lacking knowledge about what is a valid item in machinePools.

I recently switched the `default values.yaml` from MachinePools to MachineDeployment and that caused the schema to lose all information about the `machinepools` 

as long as we generate the schema from values this will of course be a problem.

the schema though is exactly the same as a machinedeployment with the addition of one field for `location`

**Comment by mogottsch** [comment]
I fixed it. With your approval, the PR is ready to be merged.

**Review by primeroz — APPROVED** [review]

**Pull request merged**